jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx

```diff
@@ -64,6 +67,7 @@ export const usePipeline = (
       ppsContext?.metadata?.config.pipeline.project?.name ?? '',
     );
     setRequirements(ppsContext?.metadata?.config.requirements ?? '');
+    setExternalFiles(ppsContext?.metadata?.config.external_files ?? '');
     setResponseMessage('');
     if (ppsContext?.metadata?.config.input_spec) {
       setInputSpec(ppsContext.metadata.config.input_spec);
```

Comment:
GH isn't letting me comment on the exact line, but I think you need to add `externalFiles` to the list of "dependencies" in the following useEffect:
```
  useEffect(() => {
    const ppsMetadata: PpsMetadata = buildMetadata();
    saveNotebookMetaData(ppsMetadata);
  }, [
    pipelineName,
    pipelineProject,
    imageName,
    requirements,
    inputSpec,
    pipelinePort,
    gpuMode,
    resourceSpec,
  ]);
```
This is the piece of code that's tracking changes and saving them to the notebook metadata.

Reply:
It appears this code is pretty brittle though, so definitely feel free to try to pitch something better if you have ideas.

Reply:
That makes sense! Yeah I might have an idea on how to refactor this slightly, but I will probably tackle it in a quick follow up PR. No need to block this on that 🤔